### PR TITLE
Fix `showrelated` command working for a Person that does not exist

### LINKED ISSUE #143: [PE-D][Tester A] When creating relation then deleting person related, the persons related still appears
When I create a relationship between two people say i/11 and i/12, 

And then I delete i/11:

and then use showrelated i/11 it gives me the relationship to i/12 still:

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: Alteqa/ped#5

### LINKED ISSUE #175: [PE-D][Tester C] Showrelated command works for a person that does not exist in address book
showrelated command for a person ID that does not exist in the address book shows 0 person listed instead of something like no person with ID 3 exists

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: kaitinghh/ped#17

### PR DESCRIPTION
**Pull request by starsia**
Fixes #175 .
Fixes #143. 

Deletion of `Person` propagates to `RelatedList` as well. 

Minor changes to improve clarity of code.

### CODE REVIEW

File: src/main/java/seedu/address/model/util/RelatedList.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public boolean hasId(IdTuple idTuple) {
      * @param idTuple The related person to be added.
      * @return True if the related person is added, false otherwise.
      */
-    public boolean remove(IdTuple idTuple) {
+    public boolean removeTuple(IdTuple idTuple) {
```
Comment (tanjieling0) [inline src/main/java/seedu/address/model/util/RelatedList.java:111]: Wrong JavaDoc?

File: src/main/java/seedu/address/model/util/RelatedList.java [hunk 2]
```diff
@@ -144,6 +144,17 @@ public List<Integer> getAllRelatedIds(RelatedList relatedList, Id id) {
         return relatedIds;
     }
 
+    /**
+     * Removes the specified ID from the relatedList.
```
Comment (tanjieling0) [inline src/main/java/seedu/address/model/util/RelatedList.java]: What is the difference between his function and the removeTuple function?

**Review by tanjieling0 — COMMENTED** [review]
Do you think we could change the message from `0 persons listed` to `Contact not found`, does it violate the feature freeze? Or otherwise could you update the UG to include this under future implementation?

Also, does this PR solve #143 instead of #175 btw

File: src/main/java/seedu/address/model/util/RelatedList.java (hunk 2 above)
Comment (starsia) [reply src/main/java/seedu/address/model/util/RelatedList.java]: The now `removeIdTuple` function removes a single `IdTuple` from the `RelatedList`, and takes in an `IdTuple`. It is used in the `unrelate` feature. The `removeId` function that I have just added takes in an `Id`, and removes **all** the `IdTuple`s with that `Id` in the `RelatedList`. It is used in the deletion of `Person`, which will result the `Person`'s `Id` to be deleted from the `RelatedList` as well.

Comment (tanjieling0) [reply src/main/java/seedu/address/model/util/RelatedList.java]: perhaps you can change the java docs to removes the tuple containing the specified ID from the related list to make it more clear?

**Review by tanjieling0 — APPROVED** [review]

**Pull request merged**